MoreLinq.Test/InterleaveTest.cs

```diff
@@ -35,6 +35,23 @@ public void TestInterleaveIsLazy()
             new BreakingSequence<int>().Interleave(new BreakingSequence<int>());
         }
 
+        /// <summary>
+        /// Verify that interleaving do not call enumerators MoveNext method eagerly
+        /// </summary>
+        [Test]
+        public void TestInterleaveDoNoCallMoveNextEagerly()
+        {
+            void Code()
+            {
+                var sequenceA = Enumerable.Range(1, 1);
+                var sequenceB = MoreEnumerable.From<int>(() => throw new TestException());
+
+                sequenceA.Interleave(sequenceB).Take(1).Consume();
+            }
+
+            Assert.DoesNotThrow(Code);
```

Comment:
This seems redundant in a test, don't you think? If an exception is thrown then the test will fail anyhow. You could then throw away the `Code` local function. If you prefer to see a test do some sort of assertion at the end then I would suggest rewriting this simply as:

```c#
[Test]
public void TestInterleaveDoNoCallMoveNextEagerly()
{
    var sequenceA = Enumerable.Range(1, 1);
    var sequenceB = MoreEnumerable.From<int>(() => throw new TestException());

    var result = sequenceA.Interleave(sequenceB).Take(1);

    Assert.DoesNotThrow(() => result.Consume());
}
```

Reply:
I simplified it in efd0f4c